### PR TITLE
clientv3,etcdserver,tools: update metrics to use promhttp

### LINKED ISSUE #8729: etcd utilizing deprecated prometheus.handler() function
As noted in a comment on prometheus.Handler:

```
// Deprecated: Please note the issues described in the doc comment of
// InstrumentHandler. You might want to consider using promhttp.Handler instead
// (which is not instrumented, but can be instrumented with the tooling provided
```

It is currently used in etcd in the following files:
```
./clientv3/example_metrics_test.go:57:		http.Serve(ln, prometheus.Handler())
./clientv3/integration/metrics_test.go:52:		srv := &http.Server{Handler: prometheus.Handler()}
./etcdserver/api/etcdhttp/metrics.go:37:	mux.Handle(pathMetrics, prometheus.Handler())
./etcdserver/api/etcdhttp/metrics.go:43:	mux.Handle(pathMetrics, prometheus.Handler())
./tools/functional-tester/etcd-tester/main.go:141:	http.Handle("/metrics", prometheus.Handler())
```

The comment recommands moving to promhttp

### PR DESCRIPTION
**Pull request by tylerauerbeck**
Deprecate prometheus.Handler() was utilized in the following files:

./clientv3/example_metrics_test.go
./clientv3/integration/metrics_test.go
./etcdserver/api/etcdhttp/metrics.go
./etcdserver/api/etcdhttp/metrics.go
./tools/functional-tester/etcd-tester/main.go

This updates etcd to utilize the replacement promhttp.Handler()

fixes #8729

### CODE REVIEW

File: etcdserver/api/etcdhttp/metrics.go [hunk 1]
```diff
@@ -23,8 +23,8 @@ import (
 	"github.com/coreos/etcd/etcdserver"
 	"github.com/coreos/etcd/etcdserver/etcdserverpb"
 	"github.com/coreos/etcd/raft"
-
-	"github.com/prometheus/client_golang/prometheus"
+
```
Comment (xiang90) [inline etcdserver/api/etcdhttp/metrics.go]: remove the extra tab. or gofmt wont pass.

**Comment by tylerauerbeck** [comment]
@xiang90 updated etcdserver/api/etcdhttp/metrics.go to remove newline. I see that some checks are still failing because it can't find the promhttp package. I'm assuming that I need to include this updated package somewhere (vendor perhaps?). I'm still picking up go, so I'm assuming that this works fine on my local machine because it's found elsewhere on my gopath. So is the solution to this to just copy the updated prometheus package from ~/go/src/github.com/prometheus to etcd/cmd/vendor/github.com/prometheus (i.e. cp -r ~/go/src/github.com/prometheus /etcd/cmd/vendor/github.com/prometheus) ?

**Comment by xiang90** [comment]
try to run ./scripts/updatedep.sh

**Comment by tylerauerbeck** [comment]
@xiang90 I may be struggling with this a little bit. I ran ./scripts/updatedep.sh like you mentioned and then pushed everything back to my branch. I then saw that there was a conflict and glide.lock. I believe I resolved that, but it looks like there are still issues as I'm seeing the same error as before in the semaphore tests and now it looks like the Jenkins tests aren't kicking off. Just let me know what I can do to help clean this up.

**Comment by gyuho** [comment]
@tylerauerbeck We do not include merge commits in PRs.
I cherry-picked your commits here https://github.com/coreos/etcd/pull/8849.

Thanks!

**Pull request closed without merging**